### PR TITLE
[#406] Publish recommendations to databus

### PR DESCRIPTION
**Pull request by wujuu**
Deployment notice. Following ENV variables need to be properly provided for each environment, **if the defaults are invalid** (`host`, `user` and `password` will 100% be invalid):

- `RS_DATABUS_HOST` - the address of your JMS provider (default: `127.0.0.1`)
- `RS_DATABUS_USERNAME` - your login to the JMS provider (default: `admin`, placeholder for the real password)
- `RS_DATABUS_PASSWORD` - your password to the JMS provider (default: `admin`, placeholder for the real password)
- `RS_DATABUS_PORT` - the port of your JMS provider (default: `61613`)
- `RS_DATABUS_SUBSCRIPTION_TOPIC` - topic on which subscriber listens to jms (default: `/topic/user_actions`)
- `RS_DATABUS_PUBLISH_TOPIC` - name of the topic to publish the recommendations on the JMS host (default: `/topic/recommendations`)
- `RS_DATABUS_SSL` - whether to use ssl when connecting to jms (default: 1) (accepted values `0` or `1`, `yes` or `no`)

Additionally the following ENV variables have been renamed:
- `RS_SUBSCRIBER_HOST -> RS_DATABUS_HOST`
- `RS_SUBSCRIBER_PORT -> RS_DATABUS_PORT`
- `RS_SUBSCRIBER_PASSWORD -> RS_DATABUS_PASSWORD`
- `RS_SUBSCRIBER_USERNAME -> RS_DATABUS_USERNAME`
- `RS_SUBSCRIBER_TOPIC-> RS_DATABUS_SUBSCRIPTION_TOPIC`
- `RS_SUBSCRIBER_SUBSCRIPTION_ID -> RS_DATABUS_SUBSCRIPTION_ID`
- `RS_SUBSCRIBER_SSL -> RS_DATABUS_SSL`
- `TEST_RS_SUBSCRIBER_HOST -> TEST_RS_DATABUS_HOST`
- `TEST_RS_SUBSCRIBER_PORT -> TEST_RS_DATABUS_PORT`
- `TEST_RS_SUBSCRIBER_USERNAME -> TEST_RS_DATABUS_USERNAME` 
- `TEST_RS_SUBSCRIBER_PASSWORD -> TEST_ RS_DATABUS_PASSWORD`
- `TEST_RS_SUBSCRIBER_TOPIC -> TEST_RS_DATABUS_SUBSCRIPTION_TOPIC`

JMS host, port, user, password and ssl are shared between the old subscription JMS cli and new JMS publisher.

Tested on local env.

### CODE REVIEW

File: recommender/tasks.py [hunk 1]
```diff
@@ -51,3 +56,31 @@ def add_user_action(user_action_raw: dict):
     """
     user_action = UserAction.parse_obj(user_action_raw)
     Deserializer.deserialize_user_action(user_action).save()
+
+
+@celery.task
+def send_recommendation_to_databus(context: dict, recommendation_response: dict):
+    host = current_app.config["RS_DATABUS_HOST"]
```
Comment (Michal-Kolomanski) [inline recommender/tasks.py:63]: Could you exclude those env variables? For example at the top of this file.

File: settings.py [hunk 2]
```diff
@@ -24,6 +24,17 @@ class Config:
     RESTPLUS_MASK_SWAGGER = False
     REDIS_HOST = f"redis://{os.environ.get('REDIS_HOST', '127.0.0.1:6379')}"
     TRAINING_DEVICE = get_device("TRAINING_DEVICE")
+    RS_DATABUS_HOST = os.environ.get("RS_DATABUS_HOST", "127.0.0.1")
+    RS_DATABUS_PORT = int(os.environ.get("RS_DATABUS_PORT", 61613))
+    RS_DATABUS_USERNAME = os.environ.get("RS_DATABUS_USERNAME", "admin")
+    RS_DATABUS_PASSWORD = os.environ.get("RS_DATABUS_PASSWORD", "admin")
+    RS_DATABUS_SSL = not os.environ.get("RS_DATABUS_SSL") in ["0", "no"]
```
Comment (Michal-Kolomanski) [inline settings.py:31]: Hmmm that is interesting. In that case, we enable SSL for everything different than ["0", "no"]. I can live with that but I wonder if that is a good practice.

**Review by Michal-Kolomanski — APPROVED** [review]
LGTM

File: settings.py (hunk 2 above)
Comment (wujuu) [reply settings.py:31]: well SSL should be enabled by default regardless - it's a good practice to have overall

File: recommender/tasks.py (hunk 1 above)
Comment (wujuu) [reply recommender/tasks.py:63]: do you mean extract?

File: docker-compose.yml [hunk 3]
```diff
@@ -16,9 +16,10 @@ services:
     - redis:/data
 
   rabbitmq:
-    image: itzg/rabbitmq-stomp:latest
+    image: rmohr/activemq
     ports:
       - ${RABBIT_HOST:-127.0.0.1:61613}:61613
+      - ${RABBIT_HOST:-127.0.0.1:8161}:8161
```
[the remark this answers is not in the document]
Comment (wujuu) [reply docker-compose.yml:22]: this is the same instance but a different port - this port on the docker container allows us to access the rabbbit web interface, while 6163 is a STOMP tcp connection port

**Comment by Michal-Kolomanski** [comment]
After merging, please update @wziajka about changes in the deployment process

**Pull request merged**